File: backend_prep/intro_to_programming/the_basics/ex_2.rb
# Introduction to Programming
# The Basics
# Exercise 2

# Use the modulo operator, division, or a combination of both to take a 4 digit number and find the digit in the:
# 1) thousands place
# 2) hundreds place
# 3) tens place
# 4) ones place

puts "Enter a 4 digit number."
num = gets.chomp.to_i

# my answer


ans_1 = num / 1000
ans_2 = (num / 100) % 10
ans_3 = (num / 10) % 10
ans_4 = num % 10

=begin

# launch school's answer

ans_1 = num / 1000
ans_2 = (num % 1000) / 100
ans_3 = (num % 100) / 10
ans_4 = num % 10

=end

puts "Value at thousands place is #{ans_1}"
puts "Value at hundreds place is #{ans_2}"
puts "Value at tens place is #{ans_3}"
puts "Value at ones place is #{ans_4}"

File: introduction_to_programming_with_javascript/the_basics/ex_2.js
// Introduction to Programming with JavaScript
// The Basics
// Exercise 2

// Use the arithmetic operators to determine the individual digits of a 4-digit number like 4936:

fourDigit = 4936;
console.log(`The number is ${fourDigit}`);
// thousands place is 4
console.log("Thousands place is: ")
console.log(parseInt(fourDigit/1000));

// hundreds place is 9
console.log("Hundreds place is: ")
console.log(parseInt((fourDigit/100)%10));

// tens place is 3
console.log("Tens place is: ")

console.log(parseInt((fourDigit/10)%100%10));

// ones place is 6
console.log("Ones place is: ")

console.log(parseInt(fourDigit%1000%100%10));